src/plugins/validation/2and3/semantic-validators/pagination-ibm.js

```diff
@@ -0,0 +1,216 @@
+                    i++
+                  ) {
+                    if (
+                      resolvedSpec.paths[head].get.parameters[i].name ===
+                        'limit' &&
+                      (resolvedSpec.paths[head].get.parameters[i].schema
+                        .type !== 'integer' ||
+                        resolvedSpec.paths[head].get.parameters[i].schema
+                          .required === true ||
+                        (!resolvedSpec.paths[head].get.parameters[i].default &&
+                          !resolvedSpec.paths[head].get.parameters[i].maximum))
+                    ) {
+                      const path = [
+                        'paths',
+                        head,
+                        'get',
+                        'parameters',
+                        i,
+                        'name'
+                      ];
+                      const message =
+                        'limit must be of type integer and must be optional with default and maximum values';
+                      paginationWarning(message, path);
+                    }
+                    if (
+                      (resolvedSpec.paths[head].get.parameters[i].name ===
+                        'start' ||
+                        resolvedSpec.paths[head].get.parameters[i].name ===
+                          'cursor') &&
+                      (resolvedSpec.paths[head].get.parameters[i].schema
+                        .type !== 'integer' ||
+                        resolvedSpec.paths[head].get.parameters[i].required ==
+                          true)
+                    ) {
+                      const message =
+                        'the start or cursor properties must be integers and optional';
+                      paginationWarning(message, path);
+                    }
+                  }
+                }
+                if (
+                  !queryParameters.includes('start') &&
+                  !queryParameters.includes('offset')
+                ) {
+                  const path = ['paths', head, 'get', 'parameters'];
+                  const message =
+                    'if start is not defined then offset must be defined and must be of type integer and optional';
+                  paginationWarning(message, path);
+                }
+                const path = [
+                  'paths',
+                  head,
+                  'get',
+                  'responses',
+                  resp,
+                  'content',
+                  'application/json',
+                  'schema',
+                  'properties',
+                  'name'
+                ];
+                nextChecker(obj, path);
+                queryLimitChecker(obj, path, queryParameters);
+                queryOffsetChecker(obj, path, queryParameters);
+                totalCountChecker(obj, path);
+                nextTokenChecker(obj, path, queryParameters);
```

Comment:
Since this batch of five methods are always called together, I don't think there is any value is making them separate methods.  Just create one method, called something like "checkResponseProperties", and do all the checks for the response properties within that method.